vocabularies/Common.xml

```diff
@@ -1454,6 +1454,132 @@ If the request succeeds the response will contain a JSON body of `Content-Type:
+        </Parameter>
+        <Parameter Name="Targets" Type="Collection(Edm.String)" Nullable="false">
+          <Annotation Term="Core.IsURL" />
+          <Annotation Term="Core.Description" String="See parameter `Targets` in [`Template_RecommendationsFunction`](#Template_RecommendationsFunction)" />
+        </Parameter>
+        <ReturnType Type="Collection(Common.PropertyRecommendationType)" Nullable="false">
+          <Annotation Term="Core.Description" String="See return type in [`Template_RecommendationsRetrievalFunction`](#Template_RecommendationsRetrievalFunction)" />
+        </ReturnType>
+      </Function>
+      <ComplexType Name="PropertyRecommendationType">
+        <Annotation Term="Common.Experimental" />
+        <Property Name="target" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.IsURL" />
+          <Annotation Term="Core.Description" String="URL (relative to the URL of the recommendations function request) addressing a property [OData-URL, section 4.6] that is targeted by this recommendation" />
+          <Annotation Term="Core.LongDescription">
+            <String>The request `SalesOrder('A')/ns.RecommendationsFunction?Targets=...`
+            might retrieve a recommendation with `"target": "Items(10)/Product"`.</String>
+          </Annotation>
+        </Property>
+        <Property Name="value" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.Description" String="Recommended value, converted to string" />
+        </Property>
+        <Property Name="text" Type="Edm.String" Nullable="true">
+          <Annotation Term="Core.Description" String="Description of the recommended value" />
+        </Property>
+        <Property Name="alternatives" Type="Collection(Common.AlternativeRecommendationType)" Nullable="false">
+          <Annotation Term="Core.Description" String="A list of alternative values, sorted by confidence score in descending order" />
+          <Annotation Term="Core.LongDescription">
+            <String>If a value is recommended via property `value`, it must be the first entry in this list.</String>
+          </Annotation>
+        </Property>
+      </ComplexType>
+      <ComplexType Name="AlternativeRecommendationType">
+        <Annotation Term="Common.Experimental" />
+        <Property Name="value" Type="Edm.String" Nullable="false">
+          <Annotation Term="Core.Description" String="Alternatively recommended value, converted to string" />
+        </Property>
+        <Property Name="score" Type="Edm.Decimal" Scale="variable" Nullable="true">
+          <Annotation Term="Core.Description" String="Confidence score of the alternatively recommended value" />
+        </Property>
+      </ComplexType>
+      <Term Name="RecommendationsRole" Type="Common.RecommendationsRoleType" Nullable="false" AppliesTo="Property Parameter">
```

Comment:
Do we need this annotation in the short-term orchestration? Does the app-specific code extension evaluate it?

Reply:
Needed.